src/core/main.c

```diff
@@ -1519,6 +1528,11 @@ static int become_shutdown(
                 /* Tell the binary how often to ping, ignore failure */
                 if (asprintf(&e, "WATCHDOG_USEC="USEC_FMT, arg_shutdown_watchdog) > 0)
                         (void) strv_push(&env_block, e);
+
+                if (arg_watchdog_device) {
+                        if (asprintf(&e, "WATCHDOG_DEVICE=%s", arg_watchdog_device) > 0)
+                                (void) strv_push(&env_block, e);
```

Comment:
This code was already buggy, so not really your fault, but please fix it if you're touching it. This should be
```C
if (asprintf(&e, ...) > 0)
        (void) strv_consume(&env_block, e);

if (arg_watchdog_device &&
    asprintf(&e, ...) > 0)
        (void) strv_consume(&env_block, e);
```